flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/LocalBufferPool.java

```diff
@@ -397,8 +394,17 @@ public void setNumBuffers(int numBuffers) throws IOException {
 			returnExcessMemorySegments();
 
 			numExcessBuffers = numberOfRequestedMemorySegments - currentPoolSize;
+
+			if (currentPoolSize > numberOfRequestedMemorySegments
```

Comment:
`currentPoolSize > numberOfRequestedMemorySegments` could be replaced by `numExcessBuffers<0`